src/providers/ZapInitProvider/Zapper.ts

```diff
@@ -56,6 +80,171 @@ export class DefaultZapper implements ZapperStrategy {
   computeMinimumMint = async (): Promise<bigint | null> => {
     return null;
   };
+
+  getSteps = (): string[] => {
+    return ['approve', 'deposit', 'transfer'];
+  };
+
+  buildContractInstructions = async (
+    oNexus: MultichainSmartAccount,
+    sendCallsExtraParams: SendCallsExtraParams,
+  ): Promise<any[]> => {
+    // Validation logic (keep this)
+    const {
+      chainId: currentChainId,
+      address: currentAddress,
+      zapData: integrationData,
+      projectData,
+    } = sendCallsExtraParams;
+
+    if (!currentChainId || !currentAddress) {
+      throw new Error('Missing chainId or address');
+    }
+
+    const currentRouteFromToken = this.currentRoute.fromToken;
+    const depositAddress = integrationData.market?.address as EVMAddress;
+    const depositToken = integrationData.market?.depositToken?.address;
+    const depositTokenDecimals = integrationData.market?.depositToken.decimals;
+    const depositChainId = projectData.chainId;
+
+    if (!depositChainId) {
+      throw new Error('Deposit chain id is undefined.');
+    }
+
+    if (!depositAddress || !depositToken) {
+      throw new Error('Deposit address or token is undefined.');
+    }
+
+    if (!depositTokenDecimals) {
+      throw new Error('Deposit token decimals is undefined.');
+    }
+
+    // @Note this works only for EVM chains
+    const isNativeSourceToken = currentRouteFromToken.address === zeroAddress;
+
+    console.warn('Using native source token:', isNativeSourceToken);
+
+    if (isNativeSourceToken) {
+      throw new Error('Native source token is not supported.');
+    }
```

Comment:
I would keep the validation logic outside separate from the `buildContractInstructions`, either in another method called before this one, or just in the `sendCalls` fn